cli/scripts/e2e/e2e.ts

```diff
@@ -48,7 +48,21 @@ for (let npmClient of ["yarn", "berry", "pnpm6", "pnpm", "npm"] as const) {
   const Suite = uvu.suite(`${npmClient}`);
 
   const repo = new Monorepo("basics");
-  repo.init(npmClient, basicPipeline);
+  repo.init(npmClient, {
+    pipeline: {
```

Comment:
Bonus test cases!

I'm trying to see what happens to the dependency graph construction when you feed it possibly-surprising input where `dependsOn` eventually contains a task that got pruned. (Not sure if this will reveal a bug in `prune` or not. 😜)

- `"#build"` (empty package name).
- Task with a `dependsOn` with a fully-qualified task id like `"c#build"` that is supposed to get pruned. 
- Task in the `dependsOn` which transitively depends on `c#build`. (AKA: do we need to build the whole graph in order to prune?)

Reply:
Upon some deeper thought, I think you are right and we will need to walk the graph. I will chalk it up to learning the codebase 😅. For whatever reason I assumed that leaving only the 'entry' packages in the turbo json would be enough but I can see now that that is not the case. I am adding a few tests and will have a new revision by the time you are up tomorrow.